src/Chapter3.hs

```diff
@@ -907,9 +977,39 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+{- ORMOLU_ENABLE -}
 class Append a where
-    append :: a -> a -> a
+  append :: a -> a -> a
+
+newtype Gold = Gold Int
 
+instance Append Gold where
+  append (Gold x) (Gold y) = Gold $ x + y
+
+-- copy pasta from previous section
+data List a
+  = Empty
+  | Cons a (List a)
+
+instance Append (List a) where
+  append x y = combine Empty x y
+    where
+      -- rebuild cons, unlike implmenting reverse in previous chapter
+      teardown :: List a -> List a -> List a
+      teardown acc Empty = acc
+      teardown acc (Cons z zs) = Cons z (teardown acc zs)
+      -- reconstruct the list, one after another
+      combine :: List a -> List a -> List a -> List a
+      combine acc Empty Empty = acc
+      combine acc xs ys = combine (teardown (teardown acc xs) ys) Empty Empty
```

Comment:
I don't quite get the logic here. You reverse lists twice, but this gives the normal order.
Append doesn't change the order of elements in List, so I think you don't need `teardown` for implementation of this function.